src/neptune_scale/sync/aggregating_queue.py

```diff
@@ -160,25 +157,27 @@ def get(self) -> BatchedOperations:
         )
 
 
-def create_run_batch(operations: list[RunOperation]) -> RunOperation:
+def create_run_batch(operations: dict[Optional[float], RunOperation]) -> RunOperation:
     if len(operations) == 1:
-        return operations[0]
+        return next(iter(operations.values()))
 
-    batch = RunOperation()
+    batch = None
+    for _, operation in sorted(operations.items(), key=lambda x: (x[0] is not None, x[0])):
+        if batch is None:
+            batch = RunOperation()
+            batch.project = operation.project
+            batch.run_id = operation.run_id
+            batch.create_missing_project = operation.create_missing_project
+            batch.api_key = operation.api_key
 
-    head_operation = operations[0]
-    batch.project = head_operation.project
-    batch.run_id = head_operation.run_id
-    batch.create_missing_project = head_operation.create_missing_project
-    batch.api_key = head_operation.api_key
-
-    for operation in operations:
         operation_type = operation.WhichOneof("operation")
         if operation_type == "update":
             batch.update_batch.snapshots.append(operation.update)
         else:
             raise ValueError("Cannot batch operation of type %s", operation_type)
 
+    if batch is None:
```

Comment:
Why this is needed?

Reply:
because our static analysis tools will notify us that def create_run_batch does not handle and empty dict as an input, even if the dict will never be empty in the currently only call site of create_run_batch